src/lib.rs

```diff
@@ -1,236 +1,217 @@
 use std::collections::HashMap;
+use std::time::{Duration, Instant};
 use std::hash::Hash;
-use std::time::Duration;
-use std::time::Instant;
+use std::fmt::Debug;
 
 #[derive(Debug, Clone)]
-struct ValueWithTimeout<V> {
-    value: V,
-    insert_time: Instant,
-    timeout: Option<Duration>,
-
+struct Entry<V> {
+  value: V,
+  insert_time: Instant
 }
 
 #[derive(Debug, Clone)]
 pub struct SimpleCache<K, V> {
-    h: HashMap<K, ValueWithTimeout<V>>,
-    timeout: Option<Duration>,
+  hashmap: Box<HashMap<K, Entry<V>>>,
```

Comment:
@giuseppe998e Why did you use a box here? I'm still learning Rust so i'm just curious :)

Reply:
Box is like malloc, so your data goes into RAM instead of CPU cache

Reply:
Okay, I knew it did allocate some space on the heap instead of the stack. That seems better. Thanks!